test/services/sift/ApplicationSiftServiceSpec.scala

```diff
@@ -167,4 +173,19 @@ class ApplicationSiftServiceSpec extends ScalaMockUnitSpec {
       whenReady(service.siftApplicationForScheme("applicationId", schemeSiftResult)) { result => result mustBe unit }
     }
   }
+
+  "sendSiftEnteredNotification" must {
+    "send email to the right candidate" in new TestFixture {
+      val candidate = CandidateExamples.minCandidate("userId")
+      val contactDetails = ContactDetailsExamples.ContactDetailsUK
+
+      (mockAppRepo.find(_ : String)).expects(appId).returningAsync(Some(candidate))
+      (mockContactDetailsRepo.find _ ).expects("userId").returningAsync(contactDetails)
+      (mockEmailClient.notifyCandidateSiftEnteredAdditionalQuestions(_: String, _: String)(_: HeaderCarrier))
+        .expects(contactDetails.email, candidate.name, *).returningAsync
+
+      whenReady(service.sendSiftEnteredNotification(appId)(new HeaderCarrier())) { result => result mustBe unit }
```

Comment:
Rather than this, I think it makes more sense here to **verify** that the function is called and the right number of times.

Reply:
Scalamock checks if a mocked method is invoked exactly once, by default. 
http://scalamock.org/user-guide/advanced_topics#call-count

I have added a test, to verify email is send only once when a candidate has multiple schemes